synapse/rest/media/v1/media_repository.py

```diff
@@ -141,14 +142,23 @@ async def create_content(
 
         Args:
             media_type(str): The content type of the file
-            upload_name(str): The name of the file
+            upload_name(str|None): The name of the file
```

Comment:
Looks like this matches [the spec](https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-media-r0-upload) which doesn't say that the upload filename is required, it also seems that the API layer event handles this properly:

https://github.com/matrix-org/synapse/blob/5cdca53aa07f921029cb8027693095d150c37e32/synapse/rest/media/v1/upload_resource.py#L57-L64